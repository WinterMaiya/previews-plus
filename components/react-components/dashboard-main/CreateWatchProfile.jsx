import axios from "axios";
import Router from "next/router";
import { useState } from "react";
import globalProfileImage from "../../helper-functions/GlobalProfileImage";
import ProfileImages from "./ProfileImages";

const CreateWatchProfile = ({ closeModal, setLoading }) => {
	// You only need to inputs to create a WatchProfile
	const [name, setName] = useState("");
	const [profileIcon, setProfileIcon] = useState();

	const submitData = async () => {
		if (!name || !profileIcon) {
			console.error("Please provide a name or select a profile icon");
			return "error";
		}
		if (!globalProfileImage.includes(profileIcon)) {
			console.error("Not a valid profile icon");
			return "error";
		}
		setLoading(true);
		closeModal();
		try {
			const create = await axios.post("/api/watchprofile/", {
				name,
				profile_pic: profileIcon,
			});
			Router.push(`/dashboard/${create.data.data.id}`);
		} catch (e) {
			console.error(e);
		}
	};

	return (
		<div className="modal-dialog modal-fullscreen text-white">
			<div className="modal-content">
				<div className="modal-header bg-dark">
					<button
						type="button"
						className="btn-close btn-close-white"
						onClick={closeModal}
					></button>
				</div>
				<div className="modal-body bg-dark bg-gradient-reverse">
					<div className="container">
						<form
							onSubmit={(e) => {
								e.preventDefault();
								submitData();
							}}
						>
							<div className="mb-3 text-center" data-testid="modal-header">
								<h2>Create New Watch Profile</h2>
							</div>
							<div className="mb-3">
								<label htmlFor="name" className="form-label" data-testid="name">
									{name ? `Hello ${name}!` : "Display Name"}
								</label>
								<input
									data-testid="input"
									type="text"
									value={name}
									className="form-control"
									id="name"
									placeholder="Display Name"
									onChange={(e) => {
										setName(e.target.value);
									}}
									required
								/>
							</div>
							<div className="mb-3 text-center">
								<button
									className="btn text-white btn-outline-primary"
									data-testid="button"
								>
									Start Watching
								</button>
							</div>
							<div className="mb-3 text-center">
								<h3 data-testid="profile-icon-header">Select A Profile Icon</h3>
								<div className="row">
									{globalProfileImage.map((e) => {
										return (
											<div className="col-md-2 col-4">
												<ProfileImages
													imageInfo={e}
													profileIcon={profileIcon}
													setProfileIcon={setProfileIcon}
												/>
											</div>
										);
									})}
								</div>
							</div>
						</form>
					</div>
				</div>
			</div>
		</div>
	);
};

export default CreateWatchProfile;
