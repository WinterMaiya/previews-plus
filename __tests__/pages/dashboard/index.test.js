import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Dashboard from "../../../pages/dashboard/index";
import CreateWatchProfile from "../../../components/react-components/dashboard-main/CreateWatchProfile";

const watchProfilesBlank = [];
const watchProfilesContent = [
	{
		id: "cl8y15biy000192a7s9qxgdms",
		name: "Test Account",
		profile_pic: "/stockphotos/racoon1.jpg",
		userId: "cl8vyy7dv000092vf9n6rwj7i",
	},
	{
		id: "cl8y15biy000192a7s9qxgdss",
		name: "Super Test Account",
		profile_pic: "/stockphotos/jellyfish1.jpg",
		userId: "cl8vyy7dv000092vf9n6rwj7i",
	},
];

describe("Dashboard Watch Profile", () => {
	it("Check Snapshot", async () => {
		const { container } = render(
			<Dashboard watchProfiles={watchProfilesContent} />
		);

		expect(container).toMatchSnapshot();
		// expect(screen.getByText("LOG OUT")).toBeInTheDocument();
	});
	it("Renders Dashboard with no watchProfiles", async () => {
		render(<Dashboard watchProfiles={watchProfilesBlank} />);

		expect(screen.getByTestId("header")).toBeInTheDocument();
	});
	it("Renders Dashboard with a watchProfile", async () => {
		render(<Dashboard watchProfiles={watchProfilesContent} />);

		expect(screen.getByTestId("header")).toBeInTheDocument();
		expect(screen.getByText("Test Account")).toBeInTheDocument();
		expect(screen.getByText("Super Test Account")).toBeInTheDocument();
	});
	it("Renders the Create Profile Modal Correctly", async () => {
		render(<CreateWatchProfile />);

		expect(screen.getByTestId("modal-header")).toBeInTheDocument();
		expect(screen.getByTestId("button")).toBeInTheDocument();
		expect(screen.getByTestId("profile-icon-header")).toBeInTheDocument();
		expect(screen.getByTestId("name")).toBeInTheDocument();
		expect(screen.getByTestId("input")).toBeInTheDocument();
	});
});
