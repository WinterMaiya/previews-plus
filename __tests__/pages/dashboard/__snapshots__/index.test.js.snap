// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Watch Profile Check Snapshot 1`] = `
<div>
  <div>
    <section
      class="container my-2 center-screen text-white"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col text-center"
          >
            <h1
              class="display-1"
              data-testid="header"
            >
              Who's Watching?
            </h1>
          </div>
        </div>
        <div
          class="row row-cols-2 row-cols-md-6"
        >
          <div
            class="col"
          >
            <div
              class="m-auto text-white"
            >
              <div
                class="my-4 watch-profile"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%27200%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Watch Profile Image"
                    class="rounded-circle card-img-top watch-profile-image"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="text-center"
              >
                <h2>
                  Test Account
                </h2>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="m-auto text-white"
            >
              <div
                class="my-4 watch-profile"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%27200%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Watch Profile Image"
                    class="rounded-circle card-img-top watch-profile-image"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="text-center"
              >
                <h2>
                  Super Test Account
                </h2>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="m-auto"
            >
              <div
                class="my-4 watch-profile"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27500%27%20height=%27500%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Watch Profile Image"
                    class="rounded-circle card-img-top watch-profile-image"
                    data-nimg="intrinsic"
                    data-testid="add-profile-con"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
          <div
            class="col d-none d-md-block"
          >
            <div
              class="m-auto"
            >
              <div
                class="my-4 watch-profile"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27500%27%20height=%27500%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Watch Profile Image"
                    class="rounded-circle card-img-top watch-profile-image"
                    data-nimg="intrinsic"
                    data-testid="add-profile-con"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
          <div
            class="col d-none d-md-block"
          >
            <div
              class="m-auto"
            >
              <div
                class="my-4 watch-profile"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27500%27%20height=%27500%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Watch Profile Image"
                    class="rounded-circle card-img-top watch-profile-image"
                    data-nimg="intrinsic"
                    data-testid="add-profile-con"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
          <div
            class="col d-none d-md-block"
          >
            <div
              class="m-auto"
            >
              <div
                class="my-4 watch-profile"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27500%27%20height=%27500%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Watch Profile Image"
                    class="rounded-circle card-img-top watch-profile-image"
                    data-nimg="intrinsic"
                    data-testid="add-profile-con"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row text-center justify-content-center"
        >
          <div
            class="col-auto"
          >
            <button
              class="btn btn-outline-secondary rounded"
            >
              <svg
                class="bi bi-pencil-square"
                fill="currentColor"
                height="25"
                viewBox="0 0 16 16"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z"
                />
                <path
                  d="M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5v11z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <span
              hidden=""
            >
              Edit
            </span>
          </div>
          <div
            class="col-auto"
          >
            <button
              class="btn btn-outline-danger rounded"
            >
              <svg
                class="bi bi-trash3"
                fill="currentColor"
                height="25"
                viewBox="0 0 16 16"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.5 1h3a.5.5 0 0 1 .5.5v1H6v-1a.5.5 0 0 1 .5-.5ZM11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3A1.5 1.5 0 0 0 5 1.5v1H2.506a.58.58 0 0 0-.01 0H1.5a.5.5 0 0 0 0 1h.538l.853 10.66A2 2 0 0 0 4.885 16h6.23a2 2 0 0 0 1.994-1.84l.853-10.66h.538a.5.5 0 0 0 0-1h-.995a.59.59 0 0 0-.01 0H11Zm1.958 1-.846 10.58a1 1 0 0 1-.997.92h-6.23a1 1 0 0 1-.997-.92L3.042 3.5h9.916Zm-7.487 1a.5.5 0 0 1 .528.47l.5 8.5a.5.5 0 0 1-.998.06L5 5.03a.5.5 0 0 1 .47-.53Zm5.058 0a.5.5 0 0 1 .47.53l-.5 8.5a.5.5 0 1 1-.998-.06l.5-8.5a.5.5 0 0 1 .528-.47ZM8 4.5a.5.5 0 0 1 .5.5v8.5a.5.5 0 0 1-1 0V5a.5.5 0 0 1 .5-.5Z"
                />
              </svg>
            </button>
            <span
              hidden=""
            >
              Delete
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
